huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go

```diff
@@ -0,0 +1,363 @@
+				Computed:    true,
+				Description: `The update time.`,
+			},
+		},
+	}
+}
+
+func resourceClientCACertificateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+
+	var (
+		createClientCACertificateHttpUrl = "v5/{project_id}/p2c-vpn-gateways/vpn-servers/{vpn_server_id}/client-ca-certificates"
+		createClientCACertificateProduct = "vpn"
+	)
+	clientCACertificateClient, err := conf.NewServiceClient(createClientCACertificateProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating VPN client: %s", err)
+	}
+
+	createClientCACertificatePath := clientCACertificateClient.Endpoint + createClientCACertificateHttpUrl
+	createClientCACertificatePath = strings.ReplaceAll(createClientCACertificatePath, "{project_id}", clientCACertificateClient.ProjectID)
+	createClientCACertificatePath = strings.ReplaceAll(createClientCACertificatePath, "{vpn_server_id}", d.Get("vpn_server_id").(string))
+
+	createClientCACertificateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	createClientCACertificateOpt.JSONBody = buildCreateClientCACertificateBodyParams(d)
+	createClientCACertificateResp, err := clientCACertificateClient.Request("POST", createClientCACertificatePath, &createClientCACertificateOpt)
+	if err != nil {
+		return diag.Errorf("error creating VPN client CA certificate: %s", err)
+	}
+
+	createClientCACertificateRespBody, err := utils.FlattenResponse(createClientCACertificateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := utils.PathSearch("client_ca_certificate.id", createClientCACertificateRespBody, "").(string)
+	if id == "" {
+		return diag.Errorf("unable to find client CA certificate ID in API response")
+	}
+	d.SetId(id)
+
+	err = waitingForClientCACertificateStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("error waiting for creating VPN client CA certificate (%s) to complete: %s", d.Id(), err)
+	}
+
+	return resourceClientCACertificateRead(ctx, d, meta)
+}
+
+func buildCreateClientCACertificateBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"client_ca_certificate": map[string]interface{}{
+			"name":    d.Get("name"),
+			"content": d.Get("content"),
+		},
+	}
+	return bodyParams
+}
+
+func waitingForClientCACertificateStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING_UPDATE"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			cfg := meta.(*config.Config)
+			region := cfg.GetRegion(d)
+
+			var (
+				httpUrl = "v5/{project_id}/vpn-servers"
+				product = "vpn"
+			)
+			client, err := cfg.NewServiceClient(product, region)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error creating CFW client: %s", err)
+			}
```

Comment:
it is unnecessary to create client for each time, it is better to be passed through param of the method